### ISSUE TITLE
VG RDF shell script much slower than VG view

### ISSUE THREAD

**Issue by JervenBolleman** [ekg/vg#111]
Investigate adding C++ code in VG view that generates appropriate RDF serialization without passing through plain JSON first. The [Turtle](http://www.w3.org/TR/turtle/) and the [Apache Thrift RDF](http://afs.github.io/rdf-thrift/) (not a standard via W3C, but very speedy) serializations are the most interesting candidates.

Existing converters can transform these serialization on demand to others of interest.

**Comment by ekg** [comment]
I'd be fine with adding support for this into vg.cpp itself. Would you hack something up to produce Turtle/RDF locally, or are you thinking of adding a more substantial library to do this?

**Comment by JervenBolleman** [comment]
I would generate turtle locally. You have to deal with only 2 datatypes, strings and integers, as well as IRIs. I would aggressively use a base and prefix declaration to minimise  the number of bytes you write out. It would be nice to have 1 extra input parameter to fill the base iri, but the default of the document iri makes a lot of sense so its not needed.

I don't know of a good C++ RDF library in any case.

**Comment by JervenBolleman** [comment]
With the resolution of ekg/vg/issues/130 we can reinvestigate JSON-LD as a serialization format.
This would need just a context description to be added before the rest of the JSON starts. This could be the minimum amount of work in the long run.

**Comment by JervenBolleman** [comment]
JSON-LD and the JSON structure currently generate do not play nicely together.
My current best result at the [JSON-LD playground](http://json-ld.org/playground/#startTab=tab-expanded&json-ld=%7B%22%40context%22%3A%7B%22sequence%22%3A%7B%22%40id%22%3A%22http%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23value%22%7D%2C%22from%22%3A%7B%22%40type%22%3A%22%40id%22%2C%22%40id%22%3A%22http%3A%2Fexample.org%2Fstep%22%7D%2C%22id%22%3A%22%40id%22%2C%22to%22%3A%7B%22%40id%22%3A%22http%3A%2F%2Fexample.org%2Fvg%2Fconnects%22%2C%22%40type%22%3A%22%40id%22%7D%2C%22node%22%3A%7B%22%40id%22%3A%22http%3A%2F%2Fexample.org%2Fvg%2Fnodes%22%2C%22%40type%22%3A%22http%3A%2F%2Fexample.org%2Fvg%2FNodeList%22%2C%22%40base%22%3A%22http%3A%2F%2Fexample.org%2Fvg%2Fnode%22%7D%2C%22edge%22%3A%22http%3A%2F%2Fexample.org%2Fvg%2Fedges%22%7D%2C%22node%22%3A%5B%7B%22sequence%22%3A%22CAAATAAG%22%2C%22id%22%3A%221%22%7D%2C%7B%22sequence%22%3A%22G%22%2C%22id%22%3A%222%22%7D%2C%7B%22sequence%22%3A%22A%22%2C%22id%22%3A%223%22%7D%5D%2C%22edge%22%3A%5B%7B%22from%22%3A%221%22%2C%22to%22%3A%222%22%7D%2C%7B%22from%22%3A%221%22%2C%22to%22%3A%223%22%7D%5D%7D) is this unsatisfactory version.
```json
{
  "@context":
  {
   
    "sequence":{"@id":"http://www.w3.org/1999/02/22-rdf-syntax-ns#value"}
    ,"from":{"@type":"@id","@id":"http:/example.org/step"}
    ,"id":"@id"
    ,"to":{"@id":"http://example.org/vg/connects","@type":"@id"}
    ,"node":{"@id":"http://example.org/vg/nodes",  "@type":"http://example.org/vg/NodeList", "@base":"http://example.org/vg/node"}
    ,"edge":"http://example.org/vg/edges"
  },
  "node": [{"sequence": "CAAATAAG", "id": "1"}, {"sequence": "G", "id":
"2"}, {"sequence": "A", "id": "3"}], "edge": [{"from": "1", "to": "2"}, {"from": "1", "to": "3"}]}
```

**Comment by JervenBolleman** [comment]
This will be fixed with pull request #171 or #170
